CalculatorEngine.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Linq;
+
+namespace GettingStartedWithCSharp
+{
+    public class CalculatorEngine : ICalculatorEngine
+    {
+        private decimal _value = 0m;
+        private decimal _memory;
+        public bool HasMemoryStored { get; private set; } = false;
+        private readonly IMessageBoxDisplayService _messageBoxDisplayService;
+        private readonly IUtils _utils;
+
+        public CalculatorEngine(IMessageBoxDisplayService messageBoxDisplayService, IUtils utils)
+        {
+            _messageBoxDisplayService = messageBoxDisplayService;
+            _utils = utils;
+        }
+
+        public decimal GetNumberToBeShown(string operation, decimal numarAfisat)
```

Comment:
Ti-am dat un nume foarte acceptabil pentru metoda asta, dar in continuare incepe cu prefixul "Get".

Ca idee, daca modifici date, e o minciuna sa ii zici "Get<ceva>", oricine se asteapta ca un "getter" (fie ca e pe o proprietate, fie ca e intr-o metoda) sa _nu modifice starea obiectului_.

Suplimentar in sine functia face altceva (mai mult) decat numele ei anunta. De ce, castiga cineva ceva daca declara fals ce face?

Deci da-i un nume care nu e mincinos.